Content.Server/Lathe/LatheSystem.cs

```diff
@@ -1,21 +1,284 @@
 using Content.Server.Lathe.Components;
+using Content.Shared.Lathe;
+using Content.Shared.Materials;
+using Content.Shared.Research.Prototypes;
+using Content.Server.Research.Components;
+using Content.Shared.Interaction;
+using Content.Server.Materials;
+using Content.Server.Popups;
+using Content.Server.Power.Components;
+using Content.Server.Stack;
+using Robust.Server.GameObjects;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Player;
+using Robust.Shared.Audio;
 using JetBrains.Annotations;
-using Robust.Shared.GameObjects;
 
 namespace Content.Server.Lathe
 {
     [UsedImplicitly]
-    internal sealed class LatheSystem : EntitySystem
+    public sealed class LatheSystem : EntitySystem
     {
+        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<LatheComponent, InteractUsingEvent>(OnInteractUsing);
+            SubscribeLocalEvent<LatheComponent, ComponentInit>(OnComponentInit);
+        }
+
+        // These queues are to add/remove COMPONENTS to the lathes
+        private Queue<EntityUid> ProducingAddQueue = new();
+        private Queue<EntityUid> ProducingRemoveQueue = new();
+        private Queue<EntityUid> InsertingAddQueue = new();
+        private Queue<EntityUid> InsertingRemoveQueue = new();
+
```

Comment:
this is kind of dog shit and i wouldnt ask you to do this but it's already coded so might as well keep it